front_end/src/app/(main)/(tournaments)/tournaments/components/tournaments_grid/live_tournament_card.tsx

```diff
@@ -0,0 +1,365 @@
+    const winnersTs = pickWinnersTs(resolvedTs, closeDate);
+
+    if (resolvedTs && nowTs >= resolvedTs) progress = 50;
+    if (winnersTs && nowTs >= winnersTs) progress = 100;
+    if (isResolved) progress = Math.max(progress, 50);
+  }
+
+  return (
+    <div>
+      <p className="my-0 hidden text-[10px] font-normal text-blue-700 dark:text-blue-700-dark lg:block">
+        {label}
+      </p>
+
+      <div className="relative mt-2 h-[4px] w-full rounded-full bg-blue-400 dark:bg-blue-400-dark">
+        <div
+          className="absolute inset-y-0 left-0 rounded-full bg-blue-700 dark:bg-blue-700-dark"
+          style={{ width: `${progress}%` }}
+        />
+
+        <ClosedChip left="0%" active />
+        <ClosedChip left="50%" active={progress >= 50} />
+        <ClosedChip left="100%" active={progress >= 100} />
+      </div>
+    </div>
+  );
+}
+
+function ClosedChip({
+  left,
+  active,
+}: {
+  left: "0%" | "50%" | "100%";
+  active: boolean;
+}) {
+  return (
+    <div
+      className={cn(
+        "absolute top-1/2 z-10 h-[14px] w-[14px] -translate-x-1/2 -translate-y-1/2 rounded-full p-[3px]",
+        active
+          ? "bg-blue-700 dark:bg-blue-700-dark"
+          : "bg-blue-400 dark:bg-blue-400-dark"
+      )}
+      style={{
+        left:
+          left === "0%" ? "5px" : left === "100%" ? "calc(100% - 5px)" : left,
+      }}
+    >
+      <div
+        className={cn(
+          "h-full w-full rounded-full",
+          active
+            ? "bg-olive-400 dark:bg-olive-400-dark"
+            : "bg-blue-400 dark:bg-blue-400-dark"
+        )}
+      />
+    </div>
+  );
+}
+
+function safeTs(iso?: string | null): number | null {
+  if (!iso) return null;
+  const t = new Date(iso).getTime();
+  return Number.isFinite(t) ? t : null;
+}
+
+function clamp01(x: number) {
+  return Math.max(0, Math.min(1, x));
+}
+
+function formatRelative(
+  t: ReturnType<typeof useTranslations>,
+  deltaMs: number
+) {
+  if (!Number.isFinite(deltaMs) || deltaMs <= 0)
+    return t("tournamentRelativeSoon");
+
+  const sec = 1000;
+  const min = 60 * sec;
+  const hour = 60 * min;
+  const day = 24 * hour;
+  const week = 7 * day;
+  const month = 30 * day;
+  const year = 365 * day;
+
+  if (deltaMs > 20 * year) return t("tournamentRelativeFarFuture");
+  if (deltaMs < min) return t("tournamentRelativeUnderMinute");
+
+  const pick = () => {
+    if (deltaMs < hour)
+      return { n: Math.round(deltaMs / min), unit: "minute" as const };
+    if (deltaMs < day)
+      return { n: Math.round(deltaMs / hour), unit: "hour" as const };
+    if (deltaMs < week)
+      return { n: Math.round(deltaMs / day), unit: "day" as const };
+    if (deltaMs < month)
+      return { n: Math.round(deltaMs / week), unit: "week" as const };
+    if (deltaMs < year)
+      return { n: Math.round(deltaMs / month), unit: "month" as const };
+    return { n: Math.round(deltaMs / year), unit: "year" as const };
+  };
```

Comment:
We have `truncateDuration` in `front_end/src/utils/formatters/date.ts` - I haven't looked very closely, but maybe it can be used here?